subsys/usb/usb_c/usbc_pe_snk_states.c

```diff
@@ -0,0 +1,839 @@
+	 *       greater than nHardResetCount the Sink Shall assume that the
+	 *       Source is non-responsive.
+	 */
+	if (atomic_test_bit(&pe->flags, PE_FLAGS_SNK_WAIT_CAP_TIMEOUT) &&
+	    pe->hard_reset_counter > PD_N_HARD_RESET_COUNT) {
+		/* Inform the DPM that the port partner is not responsive */
+		policy_notify(dev, PORT_PARTNER_NOT_RESPONSIVE);
+
+		/* Pause the Policy Engine */
+		data->pe_enabled = false;
+		return;
+	}
+
+	atomic_clear_bit(&pe->flags, PE_FLAGS_SNK_WAIT_CAP_TIMEOUT);
+	atomic_clear_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR);
+
+	/* Request the generation of Hard Reset Signaling by the PHY Layer */
+	prl_execute_hard_reset(dev);
+	/* Increment the HardResetCounter */
+	pe->hard_reset_counter++;
+}
+
+/**
+ * @brief PE_SNK_Hard_Reset Run State
+ */
+void pe_snk_hard_reset_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/*
+	 * Transition to the PE_SNK_Transition_to_default state when:
+	 *  1) The Hard Reset is complete.
+	 */
+	if (atomic_test_bit(&pe->flags, PE_FLAGS_HARD_RESET_PENDING)) {
+		return;
+	}
+
+	pe_set_state(dev, PE_SNK_TRANSITION_TO_DEFAULT);
+}
+
+/**
+ * @brief PE_SNK_Transition_to_default Entry State
+ */
+void pe_snk_transition_to_default_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	LOG_INF("PE_SNK_Transition_to_default");
+
+	/* Reset flags */
+	pe->flags = ATOMIC_INIT(0);
+	pe->data_role = TC_ROLE_UFP;
+
+	/*
+	 * Indicate to the Device Policy Manager that the Sink Shall
+	 * transition to default
+	 */
+	policy_notify(dev, SNK_TRANSITION_TO_DEFAULT);
+	/*
+	 * Request the Device Policy Manger that the Port Data Role is
+	 * set to UFP
+	 */
+	policy_notify(dev, DATA_ROLE_IS_UFP);
+}
+
+/**
+ * @brief PE_SNK_Transition_to_default Run State
+ */
+void pe_snk_transition_to_default_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/*
+	 * Wait until Device Policy Manager has transitioned the sink to
+	 * default level
+	 */
+	if (policy_is_snk_at_default(dev)) {
+		/* Inform the Protocol Layer that the Hard Reset is complete */
+		prl_hard_reset_complete(dev);
+		pe_set_state(dev, PE_SNK_STARTUP);
+	}
+}
+
+/**
+ * @brief PE_SNK_Get_Source_Cap Entry State
+ */
+void pe_snk_get_source_cap_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	LOG_INF("PE_SNK_Get_Source_Cap");
+
+	/* Send a Get_Source_Cap Message */
+	send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_GET_SOURCE_CAP);
+}
+
+/**
+ * @brief PE_SNK_Get_Source_Cap Run State
+ */
+void pe_snk_get_source_cap_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+	struct usbc_port_data *data = dev->data;
+	struct protocol_layer_rx_t *prl_rx = data->prl_rx;
+
+	/* Wait until message is sent or dropped */
```

Comment:
This is actually a state they changed recently in PD 3.1.  It used to be "send the get_source_cap, and just head back to ready", but now it runs the sender response timer and follows some trickily written "if" checks to either snk ready or snk_evaluate_capability

Reply:
I've updated the state